### ISSUE TITLE
Provide API to eagerly configure all lazy tasks

### ISSUE THREAD

**Issue by ZakTaccardi** [gradle/gradle#20496]
AFAIK, Gradle does not have an easy to use API to eagerly configure all tasks in a multi-project build.

Use case:

I have a task named: `:featureSpoonDebugAndroidTest` that groups Spoon tasks for a feature across multiple subprojects. When running it, it hits an error.

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':featureSpoonDebugAndroidTest'.
> Task with path ':feature:feature-ui:spoonDebugAndroidTest' not found in root project 'my-app'.
```

It fails because we removed the Spoon code completely, with the exception of this one task. I want CI to be able to catch this error.

Android Studio/IntelliJ has a way to configure all tasks eagerly - please expose this API to the CLI as well.

**Comment by ZakTaccardi** [comment]
Note: one problem this brings up is for tasks that are intentionally meant to fail configuration, for example - a publish task that is missing credentials. And this type of task needs to be separated from my above exception where a task depends on tasks that no longer exist is an interesting problem to solve

**Comment by Vampire** [comment]
> AFAIK, Gradle does not have an easy to use API to eagerly configure all tasks in a multi-project build.

That would probably be `allprojects { tasks.all { } }`, while `allprojects { ... }` is bad of course. :-)

**Comment by TWiStErRob** [comment]
(And if you move @Vampire's trick to an init script, you have a CLI argument that you can run only on CI ;)

**Comment by asodja** [comment]
Sorry for late reply.

This issue will be closed as a duplicate of more recent

* #32400


If you think our analysis is wrong, please provide us with more detailed information explaining why.

***


We are closing this one in favour of more recent issu which we already requalified as a documentation issue as there is already a solution.
